File: g595.py
n=int(input())
d=list(map(int,input().split()))
d.insert(0,999)
d.append(999)
ans=0
for i in range(1,n+1):
    if d[i]==0:
        if d[i-1]>d[i+1]:
            ans+=d[i+1]
        else:
            ans+=d[i-1]
print(ans)

File: g595.cpp
#include <iostream>
using namespace std;

int main()
{
    int n,ans=0;
    int h[200];
    cin >> n;
    h[0]=999;
    for (int i=1;i<=n;i++){
        cin >> h[i];
    }
    h[n+1]=999;
    for (int i=1;i<=n;i++){
        if (h[i] == 0){
            if (h[i-1] > h[i+1]){
                ans=ans+h[i+1];
            }else{
                ans=ans+h[i-1];
            }
        }
    }
    cout << ans;
}
